detection_rules/attack.py

```diff
@@ -150,3 +153,56 @@ def update_threat_map(rule_threat_map):
     for entry in rule_threat_map:
         for tech in entry['technique']:
             tech['name'] = technique_lookup[tech['id']]['name']
+
+
+def retrieve_replacement_id(asset_id: str):
+    """Get the revised ID for a revoked ATT&CK asset."""
+    if asset_id in (tactics_map.values()):
+        attack_type = 'tactics'
+    elif asset_id in list(technique_lookup):
+        attack_type = 'techniques'
+    else:
+        raise ValueError(f'Unknown asset_id: {asset_id}')
+
+    response = requests.get(f'https://attack.mitre.org/{attack_type}/{asset_id.replace(".", "/")}')
```

Comment:
I did not see anything within the existing attack data for which you could infer old to new technique ID mappings, so this was the next best way

Reply:
https://attack.mitre.org/docs/subtechniques/subtechniques-crosswalk.json

from https://github.com/elastic/detection-rules/issues/215#issue-684766124

Reply:
that is so much better! I wonder if they have that per version

Reply:
Added crosswalk but leaving this. We should only ever need this if another major change (or to validate no change)

Reply:
good thinking.